sale_commission_product_criteria/models/sale.py

```diff
@@ -0,0 +1,52 @@
+# © 2023 ooops404
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0.html
+from odoo import api, fields, models
+
+
+class SaleOrderLineAgent(models.Model):
+    _inherit = "sale.order.line.agent"
+
+    discount = fields.Float(related="object_id.discount")
+    applied_commission_item_id = fields.Many2one("commission.item")
```

Comment:
Do you need all these related fields?

Reply:
Yes. discount is needed for calculations in sale_commission_product_criteria_discount. 
applied_commission_item_id - just to show which commission item was used.